medusa/scene_numbering.py

```diff
@@ -367,36 +379,58 @@ def get_scene_numbering_for_show(indexer_id, indexer):
     return result
 
 
-def get_xem_numbering_for_show(indexer_id, indexer, refresh_data=True):
+def get_xem_numbering():
     """
-    Returns a dict of (season, episode) : (sceneSeason, sceneEpisode) mappings
-    for an entire show.  Both the keys and values of the dict are tuples.
-    Will be empty if there are no scene numbers set in xem
+    Get xem numbering.
     """
-    if indexer_id is None:
-        return {}
+    global XEM_NUMBERING
+    if XEM_NUMBERING is None:
+        main_db_con = db.DBConnection()
+        rows = main_db_con.select(
+            'SELECT indexer, indexerid, '
+            ' season, episode, '
+            ' scene_season, scene_episode '
+            'FROM tv_episodes '
+            'WHERE (scene_season or scene_episode) != 0 '
+            'ORDER BY indexer, indexerid, season, episode',
+        )
 
-    indexer_id = int(indexer_id)
-    indexer = int(indexer)
+        # Build the XEM_NUMBERING dict
+        result = defaultdict(dict)
+        for row in rows:
+            series = int(row[b'indexer']), int(row[b'indexerid'])
+            episode = int(row['season']), int(row['episode'])
+            scene = int(row['scene_season']), int(row['scene_episode'])
+            result[series][episode] = scene
 
-    if refresh_data:
-        xem_refresh(indexer_id, indexer)
+        XEM_NUMBERING = result
 
-    main_db_con = db.DBConnection()
-    rows = main_db_con.select(
-        'SELECT season, episode, scene_season, scene_episode FROM tv_episodes WHERE indexer = ? and showid = ? and (scene_season or scene_episode) != 0 ORDER BY season, episode',
-        [indexer, indexer_id])
+    return XEM_NUMBERING
 
-    result = {}
-    for row in rows:
-        season = int(row['season'])
-        episode = int(row['episode'])
-        scene_season = int(row['scene_season'])
-        scene_episode = int(row['scene_episode'])
 
-        result[(season, episode)] = (scene_season, scene_episode)
+def get_xem_numbering_for_show(indexer_id, indexer, **kwargs):
+    """
+    Returns a dict of (season, episode) : (scene season, scene episode)
+    mappings for an entire show.  Both the keys and values of the dict
+    are tuples.
 
-    return result
+    Will be empty if there are no scene numbers set in xem
+    """
+    if 'refresh_data' in kwargs:
+        warnings.warn('refresh_data is deprecated!', DeprecationWarning)
+
+    series = None, None
+    try:
+        series = int(indexer), int(indexer_id)
+    except TypeError:
+        logger.log(
+            'Unable to determine series. Indexer: {indexer}, ID: {id}'.format(
+                indexer=indexer, id=indexer_id,
+            ),
+            logger.WARNING
+        )
+    finally:
+        return get_xem_numbering()[series]
```

Comment:
This is intentional.  The primary purpose of this statement is to convert the indexer and id to integers for use as a key for the xem numbering dictionary.  If a TypeError is raised the indexer and id are not valid and thus we return the results for `series = None, None`.   Since `None, None` should not exist in the xem numbering it will return the default, an empty dict.